modules/frontend/v1/frontend.go

```diff
@@ -254,24 +284,48 @@ func (f *Frontend) Process(server frontendv1pb.Frontend_ProcessServer) error {
 			resps <- resp
 		}()
 
-		select {
-		// If the upstream request is cancelled, we need to cancel the
-		// downstream req.  Only way we can do that is to close the stream.
-		// The worker client is expecting this semantics.
-		case <-req.originalCtx.Done():
-			return req.originalCtx.Err()
-
-		// Is there was an error handling this request due to network IO,
-		// then error out this upstream request _and_ stream.
-		case err := <-errs:
-			req.err <- err
+		err = reportResponseUpstream(reqBatch, errs, resps)
```

Comment:
Do we have a context to pass?  Wondering if it might simplify the context handling below.

Reply:
If the streaming GRPC server connection itself drops or context is cancelled then the `.Send()` returns an error and this case is hit:
https://github.com/grafana/tempo/pull/2677/files#diff-0914703aed52090bd72851004df203444207d9d48677c10860b0459afef1a0b9R311

If the request is cancelled upstream then this case is hit:
https://github.com/grafana/tempo/pull/2677/files#diff-0914703aed52090bd72851004df203444207d9d48677c10860b0459afef1a0b9R304

If the requests are cancelled downstream then we get an http response and this case is hit:
https://github.com/grafana/tempo/pull/2677/files#diff-0914703aed52090bd72851004df203444207d9d48677c10860b0459afef1a0b9R304

I think everything is covered.